Extension/src/LanguageServer/copilotCompletionContextTelemetry.ts

```diff
@@ -95,6 +96,33 @@ export class CopilotCompletionContextTelemetry {
         if (maxCaretDistance !== undefined) { this.addMetric('request.maxCaretDistance', maxCaretDistance); }
     }
 
+    public addTraitsMetadata(projectContext: any, telemetryProperties: Record<string, string>, telemetryMetrics: Record<string, number>): void {
+        this.addProperty("traits.present", projectContext ? "true" : "false");
+        for (const key in telemetryMetrics) {
+            this.addMetric(`traits.${key}`, telemetryMetrics[key]);
+        }
+        for (const key in telemetryProperties) {
+            this.addProperty(`traits.${key}`, telemetryProperties[key]);
+        }
+    }
+
+    public addSnippetCollectionStatistics(statistics: CopilotCompletionContextStatistics | undefined): void {
+        if (statistics) {
+            for (const key in statistics) {
+                this.addMetric(`response.statistics.${key}`, statistics[key as keyof CopilotCompletionContextStatistics]);
+            }
+        }
+    }
+
+    public addCppStandardVersionMetadata(standardVersion: string, getProjectContextStartTime: number): void {
+        this.addProperty('response.cppStandardVersion', standardVersion);
+        this.addMetric('response.getProjectContextElapsedMs', Date.now() - getProjectContextStartTime);

```

Comment:
@lukka This telemetry is not being computed correctly: ![image](https://github.com/user-attachments/assets/94483e75-1899-427e-9a81-4912b4b1ef62)

I think you want to remove `Date.now() -`.

Reply:
yes, thanks for letting me know about this! Fixed now.